Camera.py

```diff
@@ -0,0 +1,428 @@
+
+    def onDocumentRestored(self, fpo):
+        """Callback triggered when document is restored"""
+        self.type = "Camera"
+        fpo.Proxy = self
+        self.setProperties(fpo)
+
+    def execute(self, fpo):
+        """Callback triggered on document recomputation (mandatory).
+        It mainly draws the camera representation on screen"""
+
+        size = 5
+        height = 10
+        # Square
+        poly1 = Part.makePolygon([(-size * 2, size, 0),
+                                  (size * 2, size, 0),
+                                  (size * 2, -size, 0),
+                                  (-size * 2, -size, 0),
+                                  (-size * 2, size, 0)])
+        # Triangle 1
+        poly2 = Part.makePolygon([(-size * 2, size, 0),
+                                  (0, 0, height * 2),
+                                  (-size * 2, -size, 0),
+                                  (-size * 2, size, 0)])
+        # Triangle 2
+        poly3 = Part.makePolygon([(size * 2, size, 0),
+                                  (0, 0, height * 2),
+                                  (size * 2, -size, 0),
+                                  (size * 2, size, 0)])
+        # Arrow (up direction)
+        poly4 = Part.makePolygon([(-size * 1.8, 1.2 * size, 0),
+                                  (0, 1.4 * size, 0),
+                                  (size * 1.8, 1.2 * size, 0),
+                                  (-size * 1.8, 1.2 * size, 0)])
+
+        fpo.Shape = Part.makeCompound([poly1, poly2, poly3, poly4])
+
+
+# ===========================================================================
+
+
+class ViewProviderCamera:
+    """View Provider of Camera class"""
+
+    def __init__(self, vobj):
+        vobj.Proxy = self
+        self.fpo = vobj.Object # Related FeaturePython object
+
+    def attach(self, vobj):
+        """Callback triggered when object is created/restored"""
+        self.fpo = vobj.Object
+
+    def getDisplayModes(self, _):
+        """Return a list of display modes"""
+        return ["Shaded"]
+
+    def getDefaultDisplayMode(self):
+        """Return the name of the default display mode.  It must be defined in
+        getDisplayModes."""
+        return "Shaded"
+
+    def setDisplayMode(self, mode):
+        """Map the display mode defined in attach with those defined in getDisplayModes.
+        Since they have the same names nothing needs to be done.
+        This method is optional."""
+        return mode
+
+    def getIcon(self):
+        """Return the icon which will appear in the tree view."""
+        return ":/icons/camera-photo.svg"
+
+    def setupContextMenu(self, vobj, menu):
+        """Setup the context menu associated to the object in tree view"""
+        action1 = QAction(QT_TRANSLATE_NOOP("Render", "Set GUI to this camera"),\
+                            menu)
+        QObject.connect(action1,\
+                SIGNAL("triggered()"),\
+                self.setGuiFromCamera)
```

Comment:
You don't need backslashes to do line continuation if the arguments are inside parentheses or brackets.
```py
QObject.connect(action1,
                SIGNAL("triggered()"),
                self.setGuiFromCamera)
```